lib/Dialect/FIRRTL/Transforms/LowerClasses.cpp

```diff
@@ -882,13 +891,16 @@ void LowerClassesPass::lowerClass(om::ClassOp classOp, FModuleLike moduleLike,
     auto propertyOperands = llvm::any_of(op.getOperandTypes(), [](Type type) {
       return isa<PropertyType>(type);
     });
+    bool needsClone = false;
+    if (auto instance = dyn_cast<InstanceOp>(op))
+      needsClone = shouldCreateClass(instance.getReferencedModuleNameAttr());
```

Comment:
I think you can lookup FModuleLike through InstanceGraph and probably we don't have to change the type

Reply:
This can avoid avoid an unnecessary instance graph lookup, as the module op is not required here. Querying just using the name seems better here.